connect/src/com/telenor/connect/headerenrichment/MobileDataFetcher.java

```diff
@@ -0,0 +1,140 @@
+package com.telenor.connect.headerenrichment;
+
+import android.net.Network;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+import android.text.TextUtils;
+import android.webkit.WebResourceResponse;
+
+import com.telenor.connect.ConnectSdk;
+import com.telenor.connect.WellKnownAPI;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.net.HttpURLConnection.HTTP_MOVED_PERM;
+import static java.net.HttpURLConnection.HTTP_MOVED_TEMP;
+import static java.net.HttpURLConnection.HTTP_SEE_OTHER;
+
+public class MobileDataFetcher {
+
+    private static final int MAX_REDIRECTS_TO_FOLLOW_FOR_HE = 5;
```

Comment:
I'm curious, why do we need a max for redirects?

Reply:
Not sure; moved code. The original commit included a link to the original design doc: https://kb.skunk-works.no/display/Scale/Facilitating+Header+Enrichment+over+WiFi+in+MobileConnect+%28and+Connect+ID%29+SDK